src/GitVersion.Core/SemVer/SemanticVersion.cs

```diff
@@ -24,14 +24,16 @@ public class SemanticVersion : IFormattable, IComparable<SemanticVersion>, IEqua
 
     public long Patch { get; init; }
 
+    public bool IsPreRelease => PreReleaseTag.HasTag();
+
     public SemanticVersionPreReleaseTag PreReleaseTag { get; init; }
 
     public SemanticVersionBuildMetaData BuildMetaData { get; init; }
 
     public bool IsLabeledWith(string value) => PreReleaseTag.HasTag() && PreReleaseTag.Name.IsEquivalentTo(value);
 
     public bool IsMatchForBranchSpecificLabel(string? value)
-        => PreReleaseTag.Name.Length == 0 || value is null || IsLabeledWith(value);
+        => (PreReleaseTag.Name == string.Empty && PreReleaseTag.Number is null) || value is null || IsLabeledWith(value);
```

Comment:
https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1156/

Reply:
Actually I'm not constructing any new string. Thus it would be the same. If you not mind I would like to leave it like it is (see https://stackoverflow.com/a/10250).

Reply:
it's not a big thing, just wanted to have a consistent approach, we will have to agree on an approach, put it in the .editorconfig and then the analyzer will know how to handle it

Reply:
Done